### PR TITLE
Fix ffmpeg build dependencies on clean ubuntu desktop amd64 14.04.3 or higher

### PR DESCRIPTION
**Pull request by walterav1984**
To fix commit 8fcbd65d178c051bd1ead43be5c746dad84e02b0 and form a base for building the current ffmpeg libs used on ubuntu linux so ffmpeg required changes like extra compile flags for DV/HDV could be added later with more ease and control.

PS: libiec61883 DV/HDV firewire camere support will be added in a new Pull request based using this one as base!

### CODE REVIEW

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
+#compile ffmpeg successful on ubuntu 14.04.3 & 16.04.1 amd64 using the CasparCG build requirments described here:
+#https://raw.githubusercontent.com/casparcg/Server/2.1.0/BUILDING
 
-sudo apt-get install build-essential curl tar pkg-config
+sudo apt-get -y update
+sudo apt-get -y upgrade
+sudo apt-get -y install build-essential curl tar pkg-config python2.7-dev zlib1g-dev autoconf libtool subversion libfontconfig1-dev ant default-jre default-jdk frei0r-plugins-dev libv4l-dev
```
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Are these dependencies needed onlu to **build** or are some of them also needed to **run** CasparCG?

Comment (walterav1984) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Only to "build" ffmpeg for casparcg on top of the build requirements of casparcg it self see BUILDING (build-essential cmake libxrandr-dev libjpeg-dev libsndfile1-dev libudev-dev libglu1-mesa-dev). This has been tested on to clean installs of ubuntu 14.04.3 and 16.04.1 this weekend. 

To "run" CasparCG server after building it requires libgcrypt on ubuntu 16.04.1 which is mentioned by @baltedewit #533 

To summarize the current CasparCG BUILDING requirements/packages and these FFmpeg related requirements/packages are enough to build CasparCG and ffmpeg on ubuntu 14.04.3-higher 16.04.1.

Comment (walterav1984) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Just noticed build-essential is mentioned twice, both in BUILDING casparcg and my ffmpeg build additions...

Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Ok, I don't think it is wrong that it is mentioned twice. It makes the ffmpeg-build-linux directory self sustaining in its own, which is good.

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh [hunk 2]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
+#compile ffmpeg successful on ubuntu 14.04.3 & 16.04.1 amd64 using the CasparCG build requirments described here:
+#https://raw.githubusercontent.com/casparcg/Server/2.1.0/BUILDING
 
-sudo apt-get install build-essential curl tar pkg-config
+sudo apt-get -y update
+sudo apt-get -y upgrade
```
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: I think these two lines (update and upgrade) should be done manually by the developer if needed. It is too invasive to do this automatically.

File: dependencies64/ffmpeg/ffmpeg-build-linux/build.sh [hunk 3]
```diff
@@ -448,3 +448,11 @@ CFLAGS="-I$TARGET_DIR/include -I$TARGET_DIR/usr/local/include" LDFLAGS="-L$TARGE
 	--enable-libv4l2 \
 	--enable-zlib
 make -j $jval && make install
+
```
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build.sh]: The lines below should be done manually by the developer IMHO. Consider that this build script will more often than not be used to upgrade ffmpeg, not only change build parameters. In that case the file names of the .so files will also change and need updating in modules/ffmpeg/CMakeLists.txt as well, so I think the copying/removing should be a manual step.

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh (hunk 2 above)
Comment (walterav1984) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Just leave out upgrade but update is needed to be in line with the repository.

File: dependencies64/ffmpeg/ffmpeg-build-linux/build.sh (hunk 3 above)
Comment (walterav1984) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build.sh]: Keep it manual, until the CmakeLists.txt files are updated automatically by some CasparCG --use-custom-ffmpeg build flag?

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh (hunk 2 above)
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: `dist-upgrade` is the **most** invasive option, but ``upgrade`` is also too invasive. Yes you can leave update if you want

File: dependencies64/ffmpeg/ffmpeg-build-linux/build.sh (hunk 3 above)
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build.sh]: Keep it manual until there is a need for it to be automatic. Recompiling ffmpeg is not something that will be done on a regular basis, only after careful consideration.

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh (hunk 1 above)
Comment (walterav1984) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: Warning its not self sustaining! It still needs some dependencies casparcg server build uses for instance "libdrm" which is pulled by the casparcg server mesa/x dependencies. 

If you would like to build ffmpeg 100% on its own I can do another clean ubuntu install without the casparcg depencencies and will debug build dependencies for another X hours?

File: dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh (hunk 2 above)
Comment (HellGore) [inline dependencies64/ffmpeg/ffmpeg-build-linux/build-ubuntu.sh]: It is common practice that you make the changes. These comments on lines of code is often used as a code review tool, meaning that the merge will not be done until the problems have been fixed in the pull. I haven't tried the "Start a review" button, maybe it offers are more formal way of handling it.

**Comment by HellGore** [comment]
Is your full name Walter Sonius?

**Comment by walterav1984** [comment]
Correct!

**Comment by HellGore** [comment]
Because of CHANGELOG
